### DISCUSSION TITLE
Does pip packaging preserve instantiation paths?

### DISCUSSION THREAD

**Discussion by slerman12** [facebookresearch/hydra#2641] [Q&A]
If I import my pip-packaged app in a new top-level directory, does that change all the instantiation paths to be rooted at the top-level directory?

For example, if I did

`module=path.to.my.module`

would I have to change it to:

`module=top_level_dir.path.to.my.module`?

**Comment by Jasha10** [comment]
It should not change the instantiation path; the instantiation path is supposed work the same way as python's import dotpaths.

For example, if you would write `from path.to.my.module import MyObject` in a python script, the corresponding dotpath that you'd use for an instantiated `_target_` would be `path.to.my.module.MyObject`.

Reference: https://hydra.cc/docs/1.3/advanced/instantiate_objects/overview/#instantiation-of-builtins
